nbviewer/templates/notebook.html

```diff
@@ -14,6 +13,8 @@
     {% endif %}
   {% endfor %}
 
+  {{ layout.head_js_icon("$('body').toggleClass('hide_input_cells');", "Show/Hide Inputs", "eye", "i") }}
```

Comment:
I think this is where the above code might actually go (inside the first argument (currently `"$('body').toggleClass('hide_input_cells');"`))…

I actually looked this up a bit more and think these are most of the necessary details though there are still things that need to be checked… E.g., If you can do this without explicitly iterating over all the elements that need to be toggled. 
I'll look at this tomorrow after sleeping on it. 

```javascript
for (x in $.('.input_area')){//input areas are the only things that have the class input area
   x.height(x.height()); //first call sets it, second call retrieves it, they may need to be separated
// the key insight is that is that x.height() should return the pixel height and x.height(val) sets the height to that val.
   x.toggleClass('hide_input_cell');//applies the toggle
}
```

Right now, I think you're applying 'hide_input_cell' to the entire body, but I figure this would be better to apply to the individual input areas. It also makes it a lot easier to open the way to have checkboxes in the future if can toggle individual cells. Of course, that means that the logic I wrote above won't work for the hide/show all button because then they can be toggled individually and this will just swap state. That's probably going to need more than one line though.